### PR TITLE
new api for create competition page

### PR DESCRIPTION
**Pull request by Viktorika-p**

### CODE REVIEW

File: codeArena/codeArena/urls.py [hunk 1]
```diff
@@ -25,7 +25,9 @@
     path('api/task/', include('task.urls')),
     path('api/user/', include('user.urls')),
     path('api/news/', include('news.urls')),
-    path('api/vacancies', include('vacancies.urls'))
+    path('api/vacancies', include('vacancies.urls')),
+    path('api/competition', include('competition.urls'))
```
Comment (Maksson) [inline codeArena/codeArena/urls.py]: I think you need add "/" in the end of api/vacancies and api/competition

File: codeArena/competition/serializers.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from rest_framework import serializers
+from .models import Competition
+
+
+class CompetitionSerializer(serializers.ModelSerializer):
+       class Meta:
+        model = Competition
+        fields = '__all__'
+
+
```
Comment (Maksson) [inline codeArena/competition/serializers.py:10]: Remove empty lines in this file

File: codeArena/codeArena/settings.py [hunk 3]
```diff
@@ -177,4 +177,4 @@
     'JWT_REFRESH_EXPIRATION_DELTA': datetime.timedelta(days=7),
     'JWT_SECRET_KEY': config('JWT_SECRET_KEY')
 
-}
+}
```
Comment (Maksson) [inline codeArena/codeArena/settings.py]: add last line

File: codeArena/competition/serializers.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+class CreateCompetitionSerializer(serializers.ModelSerializer):
+        class Meta:
+            model = Competition
+            fields = '__all__'
+
+
```
Comment (vvelizariy) [inline codeArena/competition/serializers.py]: remove empty line

File: codeArena/codeArena/urls.py [hunk 5]
```diff
@@ -25,7 +25,9 @@
     path('api/task/', include('task.urls')),
     path('api/user/', include('user.urls')),
     path('api/news/', include('news.urls')),
-    path('api/vacancies', include('vacancies.urls'))
+    path('api/vacancies/', include('vacancies.urls')),
+    path('api/competition/', include('competition.urls'))
```
Comment (MrSancho42) [inline codeArena/codeArena/urls.py:29]: Add coma in end of line

File: codeArena/competition/views.py [hunk 6]
```diff
@@ -1,3 +1,23 @@
...
+from .models import Competition
+from .serializers import CreateCompetitionSerializer, CompetitionSerializer
 
-# Create your views here.
+class CompetitionInfoView(APIView):
+
+    def get(self, request):
+        competition = Competition.objects.all()
+        serializer = CompetitionSerializer(competition, many=True)
+        return Response(serializer.data)
```
Comment (MrSancho42) [inline codeArena/competition/views.py:14]: Also return HTTP status

File: codeArena/competition/views.py [hunk 7]
```diff
@@ -1,3 +1,23 @@
...
+
+
+class CreateCompetitionView(APIView):
+
+    def post(self, request, format='json'):
+        competition = CreateCompetitionSerializer(data=request.data)
+        if competition.is_valid(raise_exception=True):
+            competition.save()
+        return Response(status=status.HTTP_201_CREATED)
```
Comment (MrSancho42) [inline codeArena/competition/views.py:23]: Missed empty line in end of file

**Review by s0fko — APPROVED** [review]

**Review by Maksson — APPROVED** [review]

**Review by alexandrv1111 — APPROVED** [review]

**Pull request merged**